dbms/src/Server/Server.cpp

```diff
@@ -1192,6 +1193,9 @@ int Server::main(const std::vector<std::string> & /*args*/)
     DM::SegmentReaderPoolManager::instance().init(server_info);
     DM::SegmentReadTaskScheduler::instance();
 
+    auto shared_block_schemas = std::make_shared<DM::SharedBlockSchemas>(*global_context);
+    global_context->setSharedBlockSchemas(shared_block_schemas);
```

Comment:
How about using the `init()` pattern? In this way the API surface is reduced (one does not have a chance to set it to something illegal).

```c++
global_context->initializeSharedBlockSchemas();
```